json/src/ser.rs

```diff
@@ -504,6 +504,17 @@ pub fn escape_bytes<W>(wr: &mut W, bytes: &[u8]) -> Result<()>
             b'\n' => b"\\n",
             b'\r' => b"\\r",
             b'\t' => b"\\t",
+            c @ b'\x00' ... b'\x1F' => {
+                if start < i {
+                    try!(wr.write_all(&bytes[start..i]));
+                }
+
+                try!(wr.write_fmt(format_args!("\\u{:04X}", c)));
```

Comment:
The [rustdoc for write_fmt](https://doc.rust-lang.org/std/fmt/trait.Write.html#method.write_fmt) says not to use it, use `write!` instead.